tests/components/test_librato.py

```diff
@@ -0,0 +1,113 @@
+    """ Tests homeassistant.components.librato module. """
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """ Init needed objects. """
+        self.hass = ha.HomeAssistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """ Stop down stuff we started. """
+        self.hass.stop()
+
+    def test_config_validation(self):
+        """ Ensure valid configuration is required """
+        self.assertFalse(librato_component.setup(self.hass, {'librato': {}}))
+
+    def test_setup(self):
+        """ Ensure setup returns True when proper configuration is provided """
+        self.assertTrue(librato_component.setup(self.hass, {
+                'librato': {
+                    'user': 'foo',
+                    'token': 'foo'
+                }
+            }))
+
+    @patch('librato.queue.Queue.add')
+    def test_state_submitted(self, mock_librato):
+        """ Test that known states are submitted """
+        self.assertTrue(librato_component.setup(self.hass, {
+                'librato': {
+                    'user': 'foo',
+                    'token': 'foo'
+                }
+            }))
+
+        state = ha.State('test.entity',STATE_ON)
+        mock_state_change_event(self.hass, state)
+        self.hass.pool.block_till_done()
+        self.assertEqual(1, mock_librato.call_count)
+
+    @patch('librato.queue.Queue.add')
+    def test_attributes_submitted(self, mock_librato):
+        """ Test that numerical attributes are parsed and submitted. """
+        self.assertTrue(librato_component.setup(self.hass, {
+                'librato': {
+                    'user': 'foo',
+                    'token': 'foo'
+                }
+            }))
+
+        state = ha.State(
+            'test.entity',
+            STATE_ON,
+            {
+                'battery_level': 99,
+                'another_floatish': '0.5',
+                'string': 'string',
+                'auto': 'ignored',
+                'node_id': 'also ignored'
+            })
+        mock_state_change_event(self.hass, state)
+        self.hass.pool.block_till_done()
+        self.assertEqual(3, mock_librato.call_count)
+
+    @patch('librato.queue.Queue.add')
+    def test_periodic_metrics(self, mock_librato):
+        self.assertTrue(librato_component.setup(self.hass, {
+                'librato': {
+                    'user': 'foo',
+                    'token': 'foo'
+                }
+            }))
+
+        state = ha.State(
+            'test.entity',
+            STATE_ON,
+            {
+                'battery_level': 99,
+                'another_floatish': '0.5',
+                'string': 'string',
+                'auto': 'ignored',
+                'node_id': 'also ignored'
+            })
+        mock_state_change_event(self.hass, state)
+        self.hass.pool.block_till_done()
+        self.assertEqual(3, mock_librato.call_count)
+
+        now = dt_util.utcnow()
+        for i in range(60):
```

Comment:
No need to fire 60 time changed events. Just 1 with seconds = 59 should be enough.